accepted/00000-use-dnf-for-rpm-repos.md

```diff
@@ -0,0 +1,177 @@
+- Feature Name: Use DNF for managing the RPM content
+- Start Date: 04-04-2019
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Uyuni contains components that support importing external repositories;
+collecting and retrieving information on packages, products, modules, and other content;
+and hosting repositories to be consumed by Spacewalk clients. For RPM repositories,
+it currently uses Zypper and libsolv (and previously used YUM), per
+[a previously implemented RFC](https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00047-yum-replacement.md).
+
+However, this makes it difficult to run Uyuni on Red Hat Enterprise Linux and CentOS servers
+without replacing base packages (violating subscription support terms in the RHEL case),
+and makes it even more challenging to properly support newer extensions to RPM repositories,
+such as RH/Fedora modules introduced in Fedora 30 and RHEL 8.
+
+This RFC proposes to redo the ["YUM replacement" RFC](https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00047-yum-replacement.md)
+and select to implement it by using DNF instead of Zypper.
+
+# Motivation
+[motivation]: #motivation
+
+The work done to switch to Zypper broke the ability for the Uyuni server to run properly
+on CentOS/RHEL servers because Zypper is unavailable.
+
+Previously, Uyuni used YUM (through the `yum` Python library module) for these pieces,
+which is available on CentOS/RHEL 7. However, in order to move to Python 3, Uyuni cannot
+use `yum` anymore.
+
+For expediency for the downstream SUSE Manager product delivery schedule, Uyuni is using
+Zypper for now. However, the intent for Uyuni is to be an open upstream project that
+supports more host environments than SUSE Manager does.
+
+Moreover, neither YUM nor Zypper have any reasonable awareness of how to properly
+handle [RH/Fedora modules](https://docs.pagure.org/modularity/). It's currently
+only implemented in DNF. Using DNF opens the door to developing features specific
+to handling modules in the future.
```

Comment:
Transactional servers are not special. The `transactional-update` mechanism is merely a shell script that runs zypper commands on top. If you _really_ wanted me to write a port of _that_ interface for DNF, I could. But from the Uyuni point of view, it doesn't matter, since there's nothing that it cares about beyond orchestrating command runs and reboots, which is not part of this feature.